Docker build fails part of the way through

# Opening a new issue
___

## Specify type:
- Bug

### Bug severity (if applicable):
- High'ish

## Description:
To help speed up UI development the UI team has been foregoing a Python compile and instead has been running the Docker compile for demo purposes.

It appears a question gets asked during the Docker build that appears to interrupt the process and it fails... ```Did you rename credential.sudo_password to credential.become_password (a CharField)? [y/N]```

## Bug Report

### Expected behavior:
Docker build to work without interrupts.

### Actual behavior:
The Docker build fails when it gets to the point...
```shell
...
Removing intermediate container a3c957e54136
Step 40/44 : RUN . ~/venv/bin/activate;make server-init server-static
 ---> Running in c434344c3d20
/root/venv/bin/python quipucords/manage.py makemigrations api --settings quipucords.settings
Did you rename credential.sudo_password to credential.become_password (a CharField)? [y/N] Traceback (most recent call last):
  File "quipucords/manage.py", line 34, in <module>
    execute_from_command_line(sys.argv)
  File "/root/venv/lib/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/root/venv/lib/python3.6/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/root/venv/lib/python3.6/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/root/venv/lib/python3.6/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/root/venv/lib/python3.6/site-packages/django/core/management/commands/makemigrations.py", line 177, in handle
    migration_name=self.migration_name,
  File "/root/venv/lib/python3.6/site-packages/django/db/migrations/autodetector.py", line 47, in changes
    changes = self._detect_changes(convert_apps, graph)
  File "/root/venv/lib/python3.6/site-packages/django/db/migrations/autodetector.py", line 186, in _detect_changes
    self.generate_renamed_fields()
  File "/root/venv/lib/python3.6/site-packages/django/db/migrations/autodetector.py", line 808, in generate_renamed_fields
    if self.questioner.ask_rename(model_name, rem_field_name, field_name, field):
  File "/root/venv/lib/python3.6/site-packages/django/db/migrations/questioner.py", line 197, in ask_rename
    field_instance.__class__.__name__), False)
  File "/root/venv/lib/python3.6/site-packages/django/db/migrations/questioner.py", line 88, in _boolean_input
    result = input("%s " % question)
EOFError: EOF when reading a line
make: *** [Makefile:64: server-makemigrations] Error 1
...
```

### Steps to reproduce:
1. Open terminal, change directories into the repo, and
2. OPTIONAL: Clean up and wipe the image
    ```shell
    $ docker system prune
    $ docker rmi [IMAGE ID]
    ```
3. From the root Quipucords directory run
    ```shell
      $ docker -D build . -t quipucords:latest
    ```

### Environment information:
Mac OSX

--- a/client/.env.development
+++ b/client/.env.development
@@ -10,4 +10,4 @@ REACT_APP_SCANS_SERVICE_RESULTS=http://localhost:4000/api/v1/scans/{0}/results/
 
 REACT_APP_SOURCES_SERVICE=http://localhost:4000/api/v1/sources/
 
-REACT_APP_DEBUG_MIDDLEWARE=false
+REACT_APP_AUTH_TOKEN=csrfspoof

--- a/client/.env.production
+++ b/client/.env.production
@@ -9,3 +9,5 @@ REACT_APP_SCANS_SERVICE_RESTART=/api/v1/scans/{0}/restart/
 REACT_APP_SCANS_SERVICE_RESULTS=/api/v1/scans/{0}/results/
 
 REACT_APP_SOURCES_SERVICE=/api/v1/sources/
+
+REACT_APP_AUTH_TOKEN=csrftoken

--- a/client/README.rst
+++ b/client/README.rst
@@ -37,6 +37,17 @@ To setup the mock API separately you can run::
 
     $ npm run api:mock
 
+Debugging Redux
+***************
+This project makes use of React & Redux. To enable Redux console logging, within the `client` directory, add a `.env.local` (dotenv) file with the follow line::
+
+  REACT_APP_DEBUG_MIDDLEWARE=true
+
+Once you've made the change, restart the project and console browser logging should appear.
+
+
+*Any changes you make to the `.env.local` file should be ignored with `.gitignore`.*
+
 Unit Tests
 ^^^^^^^^^^
 To run the unit tests, use this command::

--- a/client/package.json
+++ b/client/package.json
@@ -18,7 +18,7 @@
   "sassDepIncludes": "--include-path node_modules/patternfly/dist/sass --include-path node_modules/patternfly/node_modules/bootstrap-sass/assets/stylesheets --include-path node_modules/patternfly/node_modules/font-awesome-sass/assets/stylesheets",
   "scripts": {
     "api:mock": "sh ./scripts/api.sh -p 4000 -f \"$(pwd)/../docs/swagger.yml\"",
-    "api:quipucords": "sh ./scripts/api.sh -p 4000",
+    "api:quipucords": "sh ./scripts/api.sh -p 4001",
     "api:update": "npm run api:mock -- -u; npm run api:quipucords -- -u",
     "api:doc": "node ./scripts/swagger.js & open http://localhost:4040/docs/api",
     "build:css": "node-sass --include-path ./src/styles $npm_package_sassDepIncludes src/styles/entitlements.scss -o src/styles/css",
@@ -30,7 +30,7 @@
     "start:js": "run-p -l api:mock api:doc start:client",
     "start": "npm-run-all -p watch:css start:js",
     "start:client": "react-scripts start",
-    "start:quipucords-docker": "npm run api:quipucords; open http://localhost:4000/index.html",
+    "start:quipucords-docker": "npm run api:quipucords; open https://localhost:4001/index.html",
     "start:quipucords": "open http://127.0.0.1:8000/index.html; source ../venv/bin/activate; cd ../; make serve",
     "test": "run-s test:**",
     "test:lint-js": "eslint --ext=js --ext=jsx src",

--- a/client/scripts/api.sh
+++ b/client/scripts/api.sh
@@ -82,17 +82,18 @@ api()
 
   if [ -z "$(docker images | grep ^$CONTAINER' ')" ] || [ "$UPDATE" = true ]; then
     echo "Setting up Docker API container"
+    (cd ../. && make clean)
     docker build -t $CONTAINER ../.
   fi
 
   if [ -z "$(docker ps | grep $CONTAINER)" ] && [ "$UPDATE" = false ]; then
     echo "Starting API..."
-    docker run -d --rm -p $PORT:8000 --name $CONTAINER $CONTAINER >/dev/null
+    docker run -d --rm -p $PORT:443 --name $CONTAINER $CONTAINER >/dev/null
   fi
 
   if [ ! -z "$(docker ps | grep $CONTAINER)" ] && [ "$UPDATE" = false ]; then
     echo "  Container: $(docker ps | grep $CONTAINER | cut -c 1-80)"
-    echo "  API running: http://localhost:$PORT/"
+    echo "  API running: https://localhost:$PORT/"
     echo "  To stop: $ docker stop $CONTAINER"
   fi
 }

--- a/client/src/redux/store.js
+++ b/client/src/redux/store.js
@@ -3,7 +3,6 @@ import thunkMiddleware from 'redux-thunk';
 import { createLogger } from 'redux-logger';
 import promiseMiddleware from 'redux-promise-middleware';
 import reduxReducers from './reducers';
-import cookies from 'js-cookie';
 
 const hydrateStore = () => {
   // Create any initial state items based on stored data (cookies etc.)
@@ -12,15 +11,11 @@ const hydrateStore = () => {
 
 let middleware = [thunkMiddleware, promiseMiddleware()];
 
-if (process.env.NODE_ENV !== 'production') {
-  if (process.env.REACT_APP_DEBUG_MIDDLEWARE === 'true') {
-    middleware.push(createLogger());
-  }
-  cookies.set(
-    'csrftoken',
-    'xLPbEvorrDHqC9poUjUpKtJcf2kmskkwVzHpkiwqO57zD9ViBtrtqEud6llTnVPH'
-  );
-  console.warn('Warning: Loading spoof auth token.');
+if (
+  process.env.NODE_ENV !== 'production' &&
+  process.env.REACT_APP_DEBUG_MIDDLEWARE === 'true'
+) {
+  middleware.push(createLogger());
 }
 
 const store = createStore(

--- a/client/src/services/userService.js
+++ b/client/src/services/userService.js
@@ -2,7 +2,14 @@ import cookies from 'js-cookie';
 
 class UserService {
   static authorizeUser() {
-    const token = cookies.get('csrftoken');
+    // ToDo: ReEvaluate placement of this spoof for auth. Also consider using a helper function.
+    if (process.env.NODE_ENV !== 'production') {
+      cookies.set(process.env.REACT_APP_AUTH_TOKEN, 'spoof');
+
+      console.warn('Warning: Loading spoof auth token.');
+    }
+
+    const token = cookies.get(process.env.REACT_APP_AUTH_TOKEN);
 
     return new Promise((resolve, reject) => {
       if (token) {
@@ -33,10 +40,8 @@ class UserService {
     return this.whoami().then(
       response =>
         new Promise((resolve, reject) => {
-          const token = cookies.get('csrftoken');
-
-          if (token) {
-            cookies.remove('csrftoken');
+          if (response && response.userName) {
+            cookies.remove(process.env.REACT_APP_AUTH_TOKEN);
 
             return resolve({
               userName: response.userName

